### PR TITLE
Developed get student route and added postman collection

### PR DESCRIPTION
**Pull request by yashodgayashan**

### CODE REVIEW

File: BackendNew/models/students.js [hunk 1]
```diff
@@ -481,6 +481,151 @@ exports.getStudents = sendStudents => {
...
+                    .push({ dateOfStart: value.DateOfStart });
+                });
+                sql.query(
+                  selectedCompanySQL,
+                  [studentId],
+                  (err, selectedCompanies) => {
+                    if (err) {
+                      sendStudent(err, null);
+                    } else {
+                      selectedCompanies.forEach(value => {
```
Comment (hasinichathu) [inline BackendNew/models/students.js:589]: just for clarification. What happen when there is no selected company?

File: BackendNew/models/students.js [hunk 2]
```diff
@@ -602,9 +602,26 @@ exports.getStudentById = (studentId, sendStudent) => {
...
+                                  cv = value.cv;
+                                } else if (value.dateOfStart) {
+                                  stratDate = value.dateOfStart;
+                                } else if (value.selectedCompany) {
+                                  selectedCompany = value.selectedCompany;
+                                }
+                              });
+                              var modifiedSet = {
+                                cv: cv,
+                                stratDate: stratDate,
```
Comment (hasinichathu) [inline BackendNew/models/students.js:620]: startDate

File: BackendNew/models/students.js (hunk 1 above)
Comment (yashodgayashan) [reply BackendNew/models/students.js:589]: For not selected student with one batch
```
{
    "studentdata": {
        "IndexNumber": "164081J",
        "FullName": "Piyumi Dasnayaka",
        "NameWithInitials": "D A P Dasanayaka",
        "PhoneNumber": 717382928,
        "Email": "piyumimadhubashini@gmail.com",
        "Sem1GPA": 4.01,
        "Sem2GPA": 3.48,
        "Sem3GPA": 4.11,
        "Sem4GPA": 3.87,
        "SGPA": 3.87,
        "PreferedArea1": "QA Engineering",
        "PreferedArea2": "Networking",
        "PreferedArea3": "UI/UX"
    },
    "batchInfo": {
        "batch16": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "companies": [
                "CreativeSoftware",
                "Virtusa",
                "Wso2"
            ]
        }
    }
}
```

For selected student with more batches
```
{
    "studentdata": {
        "IndexNumber": "164106T",
        "FullName": "Yashod Perera",
        "NameWithInitials": "U L Y G Perera",
        "PhoneNumber": 716931384,
        "Email": "hasinisenanayaka@gmail.com",
        "Sem1GPA": 3.89,
        "Sem2GPA": 3.78,
        "Sem3GPA": 4.03,
        "Sem4GPA": 3.57,
        "SGPA": 3.72,
        "PreferedArea1": "UI/UX",
        "PreferedArea2": "Networking",
        "PreferedArea3": "QA Engineering"
    },
    "batchInfo": {
        "batch16": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "selectedCompany": "Virtusa",
            "companies": [
                "Virtusa",
                "Wso2"
            ]
        },
        "batch17": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "selectedCompany": "Wso2",
            "companies": [
                "Virtusa",
                "Wso2"
            ]
        }
    }
}
```

File: BackendNew/models/students.js [hunk 3]
```diff
@@ -602,9 +602,26 @@ exports.getStudentById = (studentId, sendStudent) => {
...
+                              var selectedCompany;
+                              val.forEach(value => {
+                                if (value.cv) {
+                                  cv = value.cv;
+                                } else if (value.dateOfStart) {
+                                  stratDate = value.dateOfStart;
+                                } else if (value.selectedCompany) {
+                                  selectedCompany = value.selectedCompany;
+                                }
+                              });
```
Comment (hasinichathu) [inline BackendNew/models/students.js:617]: Can't there be any instance where all cv, dateOfStart, selectedCompany  has value?

File: BackendNew/models/students.js (hunk 2 above)
Comment (yashodgayashan) [reply BackendNew/models/students.js:620]: @hasinichathu What do you mean by startdate?

File: BackendNew/models/students.js (hunk 3 above)
Comment (yashodgayashan) [reply BackendNew/models/students.js:617]: No. That's why I used this.

File: BackendNew/models/students.js (hunk 2 above)
Comment (yashodgayashan) [reply BackendNew/models/students.js:620]: If you asked if there is no start date. This is the sample query for that.

```
{
    "studentdata": {
        "IndexNumber": "164120F",
        "FullName": "Chamodi Sandeepani",
        "NameWithInitials": "SANDEEPANI K.G.C.",
        "PhoneNumber": 716931384,
        "Email": "chamodisandeepani@gmail.com",
        "Sem1GPA": 3.89,
        "Sem2GPA": 3.78,
        "Sem3GPA": 4.03,
        "Sem4GPA": 3.57,
        "SGPA": 3.72,
        "PreferedArea1": "UI/UX",
        "PreferedArea2": "Networking",
        "PreferedArea3": "QA Engineering"
    },
    "batchInfo": {
        "batch16": {
            "cv": "cv",
            "companies": []
        }
    }
}
```

Comment (hasinichathu) [reply BackendNew/models/students.js:620]: spellings should be startDate

File: BackendNew/models/students.js [hunk 4]
```diff
@@ -463,12 +474,304 @@ exports.getStudents = callback => {
...
+                              var modifiedSet = {
+                                cv: cv,
+                                stratDate: stratDate,
```
Comment (yashodgayashan) [inline BackendNew/models/students.js:620]:
```suggestion
                                startDate: startDate,
```

**Comment by yashodgayashan** [comment]
@hasinichathu resolved requested change for spelling mistake for startdate.

File: BackendNew/models/students.js (hunk 3 above)
Comment (hasinichathu) [reply BackendNew/models/students.js:617]: Got it.

**Pull request merged**